File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/IRightShiftRegister.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DFIR_Compiler
{
    public class IRightShiftRegister:IBorderNode
    {
        private AssociatedLeftShiftRegister associatedLeftShiftRegister;

        public IRightShiftRegister(int nodeId, int parentId, List<ITerminal> inputTerminals, List<ITerminal> outputTerminals, AssociatedLeftShiftRegister _associatedLeftShiftRegister) : base(nodeId, parentId, inputTerminals, outputTerminals)
        {
            associatedLeftShiftRegister = _associatedLeftShiftRegister;
            INodeType = "IRightShiftRegister";
        }

        public virtual AssociatedLeftShiftRegister GetAssociatedLeftShiftRegister()
        {
            return associatedLeftShiftRegister;
        }

        public override void Print()
        {
            Console.WriteLine("------------------------------------");
            Console.WriteLine("IRightShiftRegister:");
            Console.WriteLine("NodeId: " + nodeId + ' ' + "ParentId: " + parentId);
            associatedLeftShiftRegister.Print();
            for (int i = 0; i < inputTerminals.Count; i++)
            {
                inputTerminals[i].Print();
            }
            for (int i = 0; i < outputTerminals.Count; i++)
            {
                outputTerminals[i].Print();
            }
        }

    }
}

File: Wayne/WayneCompiler/src/Dfir_representation/IRightShiftRegister.java
package Dfir_representation;

import java.util.Vector;


public class IRightShiftRegister extends IBorderNode{
	   
	   private AssociatedLeftShiftRegister associatedLeftShiftRegister;
	   public IRightShiftRegister(int nodeId, int parentId, Vector<ITerminal> inputTerminals,Vector<ITerminal> outputTerminals, AssociatedLeftShiftRegister associatedLeftShiftRegister){
		   super(nodeId,parentId,inputTerminals,outputTerminals);
		   this.associatedLeftShiftRegister = associatedLeftShiftRegister;
		   JNodeType = "IRightShiftRegister";
	   }
	   public AssociatedLeftShiftRegister getAssociatedLeftShiftRegister(){
		   return this.associatedLeftShiftRegister;
	   }
	   
	   public void print(){
		   System.out.println("IRightShiftRegister begin");
		   System.out.println("NodeId: " + this.nodeId + ' ' + "ParentId: " + this.parentId);
		   this.associatedLeftShiftRegister.print();
		   for(int i=0; i<inputTerminals.size(); i++){
				   inputTerminals.get(i).print();
		   }
		  for(int i=0; i<outputTerminals.size(); i++){
				   outputTerminals.get(i).print();
		   }   
		  System.out.println("IRightShiftRegister end\n");
	   }
}